### PR TITLE
Dave/nav bar loggedin

### PR DESCRIPTION
**Pull request by Serranoda**
This PR includes the following:

Navbar.tsx:

Includes the navigation links for logged in users and a drop down for user to go to their profile and or logout

NavbarMenuDropdown.tsx:

Includes logic for dropdown menu. Dropdown menu is from MUI in which I modified to fit our style.

LandingNavbar.tsx changes:

I made changes to the LandingNavbar to make its appearance consistent with the logged in Navbar. Very simple spacing changes

### CODE REVIEW

File: client/src/ui/NavBar.tsx [hunk 1]
```diff
@@ -1,44 +1,154 @@
...
+import { usePathname } from "next/navigation";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faMessage,
+  faChartLine,
+  faMagnifyingGlass,
+} from "@fortawesome/free-solid-svg-icons";
+import { NavbarMenuDropdown } from "./NavbarMenuDropdown";
 
-interface ILinks {
```
Comment (MekhiHart) [inline client/src/ui/NavBar.tsx:4]: Why did you remove this? I've already set it up to make the NavBar easier to change?

**Review by MekhiHart — CHANGES_REQUESTED** [review]

File: client/src/ui/NavBar.tsx (hunk 1 above)
Comment (Serranoda) [reply client/src/ui/NavBar.tsx:4]: I was conforming to the already merged landingnavbar

Comment (Serranoda) [reply client/src/ui/NavBar.tsx:4]: reinstated the array and use mapping to generate links for navigation

File: client/src/ui/NavBar.tsx [hunk 2]
```diff
@@ -1,44 +1,111 @@
...
   href: string;
+  icon: JSX.Element;
 }
 export function NavBar() {
   const links: ILinks[] = [
     {
       name: "Explore",
       href: "/explore",
+      icon: (
+        <FontAwesomeIcon
```
Comment (MekhiHart) [inline client/src/ui/NavBar.tsx:25]: I wouldn't store the actual component. What you can do instead is add the Icon (ex: icon: faMagnifyingGlass). Then you map it over the actual FontAwesomeIcon component. This reduces redundancy of creating the each component in the array

File: client/src/ui/LogoNameLink.tsx
Comment (MekhiHart) [inline client/src/ui/LogoNameLink.tsx:1]: Place this under ui/landingpage if this component is only being used in the Landingpage

File: client/src/ui/NavbarMenuDropdown.tsx
Comment (MekhiHart) [inline client/src/ui/NavbarMenuDropdown.tsx:1]: Place this under ui/landingpage if this component is only being used in the Landingpage

**Review by MekhiHart — CHANGES_REQUESTED** [review]
Looks good besides the array containing the component. Nice job

File: client/src/ui/NavBar.tsx [hunk 3]
```diff
@@ -1,44 +1,111 @@
...
+      {/* Container for logo and navigation links. Note: This is to align these element to left while the drop down is aligned to the right  */}
+      <div
+        style={{
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+        }}
+      >
+        {/* Logo Section Note: Links to explore page because that will be our users homepage */}
+        <LogoNameLink href="/expore" />
```
Comment (MekhiHart) [inline client/src/ui/NavBar.tsx:69]: Fix this to link to the "/explore" route

**Review by MekhiHart — CHANGES_REQUESTED** [review]

**Review by MekhiHart — APPROVED** [review]
LGTM
